src/geometry/cached_system.hpp

```diff
@@ -409,6 +428,18 @@ void InitializeCachedCoordinateSystem(ParameterInput *pin, StateDescriptor *geom
     }
     geometry->AddField("g.c." + var_names[i], m);
   }
+  // Do metric derivative and lapse derivative separately, as
+  // they're only available on cell centers
+  for (int i = 0; i < cent_only_sizes.size(); ++i) {
+    if (axisymmetric) {
+      shape = {dims.nx1, dims.nx2, 1, cent_only_sizes[i]};
+      m = Metadata(flags_o, shape);
+    } else {
+      shape = {cent_only_sizes[i]};
```

Comment:
Should this not be `shape = {dims.nx1, dims.nx2, dims.nx3, cent_only_sizes[i]};`?

Reply:
No, it can be just `cent_only_sizes[i]`. If you only pass in 1 number, and tell Parthenon that your variable is cell-centered, it infers the other dimension.

Reply:
Ah right, thanks :+1: